.github/workflows/CD.yml

```diff
@@ -12,6 +14,9 @@ on:
       - test
       - main
 
+env:
+  ENV: ${{ github.ref == 'refs/heads/dev' && 'd' || 't' }}
```

Comment:
Basicly that does work in this context, but is little bit scary if anything else changes or trusts in ENV var. I would add check that t is only if test branch used.